src/wasm/wasm-binary.cpp

```diff
@@ -2233,16 +2254,47 @@ void WasmBinaryReader::readHeader() {
   }
 }
 
-void WasmBinaryReader::readStart() { startIndex = getU32LEB(); }
+void WasmBinaryReader::readStart() {
+  startIndex = getU32LEB();
+  wasm.start = getFunctionName(startIndex);
+}
 
 static Name makeName(std::string prefix, size_t counter) {
   return Name(prefix + std::to_string(counter));
 }
 
+// Look up a name from the names section or use a validated version of the
+// provided name. Return the name and whether it is explicit in the input.
+static std::pair<Name, bool>
+getOrMakeName(const std::unordered_map<Index, Name>& nameMap,
+              Index i,
+              Name name,
+              std::unordered_set<Name> usedNames) {
```

Comment:
```suggestion
              std::unordered_set<Name>& usedNames) {
```
Otherwise new names we add are not noted as used?

Reply:
Oops! That's important.